src/errors.js

```diff
@@ -4,35 +4,75 @@
  * Exception raised when trying to revert migration that is not possible
  * to revert.
  */
-class NonReversibleMigration extends Error {
+class NonReversibleMigrationError extends Error {
   constructor (message) {
     super(message)
-    this.name = 'NonReversibleMigration'
+    this.name = 'NonReversibleMigrationError'
+    this.code = 'ERR_NON_REVERSIBLE_MIGRATION'
     this.message = message
   }
 }
-exports.NonReversibleMigration = NonReversibleMigration
+
+NonReversibleMigrationError.code = 'ERR_NON_REVERSIBLE_MIGRATION'
```

Comment:
If you want to assert that some error is being thrown with an error code of the error, then yes you need it because if it is declared only in constructor then you don't have any way how to retrieve without instantiating it and hence `if (err.code == errors.NonReversibleMigrationError.code) ...` not possible.

Reply:
Btw. this is pretty much same approach that is proposed by @hugomrdias here: https://github.com/ipfs/js-ipfs/pull/1746/files#diff-00052fb38502dc1ed2fbc79ca57f9e4e

Reply:
Ok makes sense 👍